src/SerilogTracing/TracingConfiguration.cs

```diff
@@ -69,47 +78,85 @@ public IDisposable TraceToSharedLogger()
 
     IDisposable EnableTracing(Func<ILogger> logger)
     {
-        var instrumentors = Instrument.GetInstrumentors().ToArray();
-        var activityListener = new ActivityListener();
-        var diagnosticListenerSubscription = DiagnosticListener.AllListeners.Subscribe(new DiagnosticListenerObserver(instrumentors));
-        var disposeProxy = new DisposeProxy(diagnosticListenerSubscription, activityListener);
-        
         ILogger GetLogger(string name)
         {
             var instance = logger();
-            return !string.IsNullOrWhiteSpace(name) ? instance.ForContext(Constants.SourceContextPropertyName, name) : instance;
+            return !string.IsNullOrWhiteSpace(name)
+                ? instance.ForContext(Constants.SourceContextPropertyName, name)
+                : instance;
         }
 
-        Sample.ConfigureSampling(activityListener);
-        
-        // Note, this will not be reevaluated if the minimum level dynamically changes.
-        activityListener.ShouldListenTo = source => source.Name == Core.Constants.SerilogActivitySourceName || GetLogger(source.Name).IsEnabled(LogEventLevel.Fatal);
+        var activityListener = new ActivityListener();
+        var disposeProxy = new DisposeProxy(activityListener,
+            DiagnosticListener.AllListeners.Subscribe(
+                new DiagnosticListenerObserver(Instrument.GetInstrumentors().ToArray())));
+
+        var levelMap = InitialLevel.GetOverrideMap();
+
+        // NOTE: We may want a config setting to only listen to specific sources
+        activityListener.ShouldListenTo = _ => true;
+
+        var sample = Sample.ActivityContext;
+        var usingParentId = Sample.ParentId;
+        activityListener.Sample = (ref ActivityCreationOptions<ActivityContext> activity) =>
+        {
+            if (!GetLogger(activity.Source.Name)
+                    .IsEnabled(GetInitialLevel(levelMap, activity.Source.Name)))
+                return ActivitySamplingResult.None;
+
+            return sample?.Invoke(ref activity) ?? ActivitySamplingResult.AllData;
+        };
+        activityListener.SampleUsingParentId = (ref ActivityCreationOptions<string> activity) =>
```

Comment:
We should only set this if the user didn't supply `UsingParentId`, otherwise it will be ignored.